main.go

```diff
@@ -686,49 +705,219 @@ func (c *coremarkAnalyzer) Close() (err error) {
 		}
 		fmt.Fprintf(f, "%s\n", strings.Join(fields, ","))
 	}
-	return
+	return nil
+}
+
+const netCSVHeader = "Cloud,Date,MachineType,Throughput,ThroughputUnits,minLat," +
+	"meanLat,p90Lat,p99Lat,maxLat,latStdDev,txnRate"
+
+type networkResult struct {
+	throughput      float64
+	throughputUnits string
+	minLat          float64
+	meanLat         float64
+	p90Lat          float64
+	p99Lat          float64
+	maxLat          float64
+	latStdDev       float64
+	txnRate         float64
+	modtime         time.Time
+}
+
+// networkResult implements the analysisResult interface.
+func (n *networkResult) ModificationTime() time.Time {
+	return n.modtime
 }
 
-const netCSVHeader = ``
+var _ analysisResult = &networkResult{}
 
 type netAnalyzer struct {
+	baseAnalyzer
 }
 
-func newNetAnalyzer() *netAnalyzer {
-	return &netAnalyzer{}
+func newNetAnalyzer(cloud string) *netAnalyzer {
+	return &netAnalyzer{
+		baseAnalyzer: baseAnalyzer{
+			cloud:          cloud,
+			machineResults: make(map[string]analysisResult),
+		},
+	}
 }
 
-func (n *netAnalyzer) Close() error {
-	log.Print("net: Close() unimplemented\n")
+func (n *netAnalyzer) Close() (err error) {
+	f, err := os.OpenFile(ResultsFile("net.csv", n.cloud), os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		return err
+	}
+	defer func() { err = f.Close() }()
+
+	fmt.Fprintf(f, "%s\n", netCSVHeader)
+	for machineType, res := range n.machineResults {
+		result := res.(*networkResult)
+		fields := []string{
+			n.cloud,
+			result.ModificationTime().String(),
+			machineType,
+			fmt.Sprintf("%f", result.throughput),
+			result.throughputUnits,
+			fmt.Sprintf("%f", result.minLat),
+			fmt.Sprintf("%f", result.meanLat),
+			fmt.Sprintf("%f", result.p90Lat),
+			fmt.Sprintf("%f", result.p99Lat),
+			fmt.Sprintf("%f", result.maxLat),
+			fmt.Sprintf("%f", result.latStdDev),
+			fmt.Sprintf("%f", result.txnRate),
+		}
+		fmt.Fprintf(f, "%s\n", strings.Join(fields, ","))
+	}
+	return nil
+}
+
+func (n *netAnalyzer) analyzeNetwork(cloud CloudDetails, machineType string) error {
+	glob := path.Join(cloud.LogDir(), FormatMachineType(machineType), "netperf-results.*/success")
+	goodRuns, err := filepath.Glob(glob)
+	if err != nil {
+		return err
+	}
+
+	for _, r := range goodRuns {
+		// Read the netperf-results
+		log.Printf("Analyzing %s", r)
+		info, err := os.Stat(r)
+		if err != nil {
+			return err
+		}
+		if res, ok := n.machineResults[machineType]; ok && res.ModificationTime().Sub(info.ModTime()) > 0 {
```

Comment:
I noticed that time has time.After -- it's better to use that vs subtraction.

Reply:
Switched it. Thanks!